tests/testthat/test-create_query_data.R

```diff
@@ -578,3 +578,26 @@ test_that("format.basket_select Test 23: formatting is correct (name specified)"
     "basket_select(name = \"My SDG\", id = NULL, scope = \"NA\", type = \"sdg\")"
   )
 })
+
+# assert_db_requirements ----
+## Test 24: assert_db_requirements_
+test_that("assert_db_requirements 24: assert_db_requirements() error if no `fun` provided", {
+  # create a query for a SMQ
+  q <- query(
+    prefix = "SMQ02",
+    id = auto,
+    definition = basket_select(
+      name = "Pregnancy and neonatal topics (SMQ)",
+      scope = "NARROW",
+      type = "smq"
+    )
+  )
+  validate_query(q)
+  queries <- list(q)
+
+  expect_snapshot(
+    assert_db_requirements(fun = NULL,
+                           fun_arg_name = "NULL", version = NULL, queries = queries, i = 1),
```

Comment:
I think it is better to use
```suggestion
    create_query_data(queries = queries),
```
and move the test into the `create_query_data` section.
Then it ensured that the correct argument name is stated if users miss to specify `get_terms_fun`. (They won't call `assert_db_requirements()`.)

Reply:
@bundfussr  Don't quite understand, so I need time to understand more of the code.

Reply:
@bundfussr 
 Looked at this again, but do not understand enough to be useful.   
Possible you could finish the PR??
Apologies.